Resolve bugzilla #1212 and retire CDATE from system

KateFriedman-NOAA opened the pull request:
# Description

This PR resolves bugzilla 1212 (#293), removes `CDATE` in all but a few places, and makes other general cleanup/maintenance changes:

1. Add copy of `global_cycle.sh` into workflow and update it to remove `CDATE` and some other EE2 changes.
2. Remove `CDATE` from system and replace with `PDYcyc` as needed. Several exceptions: 1) add `CDATE` into `jobs/JGFS_ATMOS_VERIFICATION` until `CDATE` is removed from EMC_verif-global and 2) leave in xml for GDASApp jobs that still need it. Left a `TODO` in `tasks.py` to remove it later.
3. General cleanup of standard environmental variables from lower scripts (bugzilla 1212).
4. Remove `SEND` variables in lower scripts. Add `SENDAWIP` to `config.base` after removing it from lower script.
5. Move tracker variables up to `config.tropcy`.
6. Remove `model` variable and replace with `RUN` as appropriate.
7. Some additional random cleanup and syntax updating.

Resolves #293 
Refs #3788

# Type of change

Bugzilla resolution and maintenance

# Change characteristics

- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO

# How has this been tested?

Cycled test.

KateFriedman-NOAA committed:
Script cleanup

- remove instances where SEND* variables are set unneccessarily
- remove model variable and replace with RUN where appropriate
- replace some CDATEs with PDYcyc
- some general syntax cleanup

Refs #293

KateFriedman-NOAA committed:
Remove CDATE instances

Refs #293

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1212

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1212

KateFriedman-NOAA committed:
Remove DATA setting in ush/gaussian_sfcanl.sh

Refs #293

KateFriedman-NOAA committed:
Remove nwprod reference in scripts/exgfs_wave_nawips.sh

Refs #293

KateFriedman-NOAA committed:
CDATE removal updates

Refs #3788

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/develop' into feature/bugzilla_1212

* origin/develop:
  Provide users the ability to customize their account/space specifics (#3769)

KateFriedman-NOAA committed:
Add SENDAWIP into config.base

Refs #293

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1212

KateFriedman-NOAA committed:
Temporarily add CDATE setting into JGFS_ATMOS_VERIFICATION

Will remove when work is done to remove CDATE in the
EMC_verif-global package. Need to define CDATE in the
workflow JJOB script for now since it is no longer defined
in the environment via the xml.

Refs #293
Refs #3788

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1212

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1212

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1212

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1212

KateFriedman-NOAA committed:
Add shellcheck disable in scripts/exglobal_diag.sh

Refs #293

aerorahul reviewed (dismissed):
looks good.  approved pending testing passes.

KateFriedman-NOAA committed:
Add global_cycle script from UFS_UTILS

- Add copy of global_cycle.sh from UFS_UTILS.
- Update global_cycle.sh for EE2 and COM variables.

Refs #3788
Refs #293

KateFriedman-NOAA committed:
Address shellcheck warnings

Refs #293

KateFriedman-NOAA committed:
Add CDATE back to xml for now

- remove when GDASApp removes CDATE

Refs #3788

KateFriedman-NOAA committed:
Address shellcheck warnings

Refs #293

KateFriedman-NOAA committed:
Address pynorms warning

Refs #293

KateFriedman-NOAA committed:
Address shellcheck warning

Refs #293

KateFriedman-NOAA commented:
Running CI suite test on WCOSS2-Cactus before opening for review.

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/bugzilla_1212

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/bugzilla_1212

KateFriedman-NOAA commented:
Testing is going well so far on Cactus. Will move this out of draft while testing continues.

DavidHuber-NOAA commented on dev/workflow/rocoto/tasks.py:
```diff
@@ -71,7 +71,7 @@ def __init__(self, app_config: AppConfig, run: str) -> None:
                       'NET': self._base.get('NET'),
                       'RUN': self.run,
-                      'CDATE': '<cyclestr>@Y@m@d@H</cyclestr>',
+                      'CDATE': '<cyclestr>@Y@m@d@H</cyclestr>',  # TODO: remove CDATE
```
Is this sticking around for the GDASApp?

KateFriedman-NOAA replied:
Yes, please remove this line when GDASApp is updated to no longer use the CDATE variable within. I saw enough instances to warrant leaving this for now.

DavidHuber-NOAA commented on scripts/exgfs_atmos_gempak_gif_ncdc_skew_t.sh:
```diff
@@ -48,7 +48,7 @@ export RSHPDY="${PDY:4:}${PDY:2:2}"
 cpreq "${HOMEgfs}/gempak/dictionaries/metar.tbl" metar.tbl
 sort -k 2n,2 metar.tbl > metar_stnm.tbl
-cpreq "${COMIN_OBS}/${model}.${cycle}.adpupa.tm00.bufr_d" fort.40
+cpreq "${COMIN_OBS}/${RUN}.${cycle}.adpupa.tm00.bufr_d" fort.40
```
Should this be `NET` instead of `RUN`?

DavidHuber-NOAA replied:
Looking around at other cases, I think `RUN` is actually correct.  Disregard.

DavidHuber-NOAA commented on scripts/exgfs_atmos_goes_nawips.sh:
```diff
@@ -48,7 +48,7 @@ pdsext=no
 
 GEMGRD="${RUN2}_${PDY}${cyc}f${fhr3}"
-GRIBIN="${COMOUT_ATMOS_GOES}/${model}.${cycle}.${GRIB}${fhr3}${EXT}"
+GRIBIN="${COMOUT_ATMOS_GOES}/${RUN}.${cycle}.${GRIB}${fhr3}${EXT}"
```
`NET`?

DavidHuber-NOAA replied:
Disregard.

DavidHuber-NOAA commented on scripts/exglobal_atmos_products.sh:
```diff
@@ -1,8 +1,5 @@
 #! /usr/bin/env bash
 
-# Programs used
-export WGRIB2=${WGRIB2:-${wgrib2_ROOT}/bin/wgrib2}
-
```
Is this getting removed because it is in the module files?

ChristopherHill-NOAA replied:
This comment may be relevant to setting WGRIB2 through module files on all systems:
https://github.com/NOAA-EMC/global-workflow/blob/3145045e4689fcea30334c936a68978788536de3/modulefiles/module_base.orion.lua#L39-L40

KateFriedman-NOAA replied:
Yes and because variables like this should not be set in lower-level scripts.

DavidHuber-NOAA commented on ush/global_cycle.sh:
```diff
@@ -0,0 +1,353 @@
+export PGMOUT=${PGMOUT:-${pgmout:-'&1'}}
+export PGMERR=${PGMERR:-${pgmerr:-'&2'}}
+export REDOUT=${REDOUT:-'1>'}
+export REDERR=${REDERR:-'2>'}
```
@GeorgeGayno-NOAA actually just reworked this in UFS_Utils as these are not being interpreted as expected.

@KateFriedman-NOAA could you update this version of `global_cycle.sh` to capture the fix in https://github.com/ufs-community/UFS_UTILS/pull/1072?

DavidHuber-NOAA reviewed (changes requested):
Looks good overall, just a couple of minor suggestions.

BoCui-NOAA commented:
@KateFriedman-NOAA 

https://github.com/KateFriedman-NOAA/global-workflow/blob/c5ff1d2c3e26b3471640fa1b227e14af8bf7ad64/ush/gfs_bfr2gpk.sh#L54

Please change
if [[ ${SENDDBN} == "YES" ]]; then
to
if [[ "${SENDDBN}" == "YES" ]]; then

https://github.com/KateFriedman-NOAA/global-workflow/blob/c5ff1d2c3e26b3471640fa1b227e14af8bf7ad64/ush/gfs_sndp.sh#L56

Please change
if [[ ${SENDDBN} == 'YES' ]] ; then
to
if [[ "${SENDDBN}" == "YES" ]] ; then

Thanks!

ChristopherHill-NOAA reviewed (dismissed):
Changes that were made to the following files appear to be well-formed:
```
jobs/JGDAS_ATMOS_GEMPAK
jobs/JGFS_ATMOS_AWIPS_20KM_1P0DEG
jobs/JGFS_ATMOS_FBWIND
jobs/JGFS_ATMOS_GEMPAK
jobs/JGFS_ATMOS_GEMPAK_META
jobs/JGFS_ATMOS_GEMPAK_NCDC_UPAPGIF
jobs/JGFS_ATMOS_GEMPAK_PGRB2_SPEC
scripts/exgdas_atmos_nawips.sh
scripts/exgfs_atmos_awips_20km_1p0deg.sh
scripts/exgfs_atmos_gempak_gif_ncdc_skew_t.sh
scripts/exgfs_atmos_goes_nawips.sh
scripts/exgfs_atmos_nawips.sh
```

WenMeng-NOAA reviewed (dismissed):
The updates related to ATMOS Product generation part look good to me.

KateFriedman-NOAA committed:
Updates to global_cycle.sh

Refs #293

DavidHuber-NOAA commented on ush/global_cycle.sh:
```diff
@@ -0,0 +1,347 @@
+ /
+EOF
+
+${APRUNCY} ${CYCLEXEC} 1>${PGMOUT} 2>${PGMERR}
```
```suggestion
"${APRUNCY}" "${CYCLEXEC}" 1>"${PGMOUT}" 2>"${PGMERR}"
```

DavidHuber-NOAA committed:
Update ush/global_cycle.sh

DavidHuber-NOAA reviewed (dismissed):
Looks good to me.  Thanks Kate!!

DavidHuber-NOAA committed:
Merge branch 'develop' into feature/bugzilla_1212

DavidHuber-NOAA commented:
Launching CI on C6.

DavidHuber-NOAA commented:
And wcoss2.

DavidHuber-NOAA commented:
All tests passed on WCOSS2.  Setting label.

DavidHuber-NOAA commented:
Correction.  I was looking at the wrong EXPDIR.  The `*sfcanl` and `enkfgdas_esfc` jobs failed on WCOSS2.  Inspecting.

DavidHuber-NOAA commented on ush/global_cycle.sh:
```diff
@@ -0,0 +1,347 @@
+ /
+EOF
+
+"${APRUNCY}" "${CYCLEXEC}" 1>"${PGMOUT}" 2>"${PGMERR}"
```
My bad.  `APRUNCY` should not be double-quoted.

```suggestion
${APRUNCY} "${CYCLEXEC}" 1>"${PGMOUT}" 2>"${PGMERR}"
```

DavidHuber-NOAA committed:
Update ush/global_cycle.sh

DavidHuber-NOAA commented:
I am kicking off Gaea from where it left off.  I will relaunch WCOSS2 manually in my own workspace.

DavidHuber-NOAA commented:
All tests completed successfully on C6 and WCOSS2.  Merging.

DavidHuber-NOAA committed:
Merge branch 'develop' into feature/bugzilla_1212
